### PR TITLE
Add timeouts

### PR DESCRIPTION
**Pull request by ipa-vsp**
This PR:

- [x] Timeout for booting all the slaves. 
- [x] Get a non-transmit timeout from bus.yml

### CODE REVIEW

**Comment by ipa-vsp** [comment]
#194

File: canopen/sphinx/user-guide/configuration.rst [hunk 1]
```diff
@@ -111,6 +111,7 @@ but come from the lely core library. Below you find a list of possible configura
   reset_all_nodes;	Specifies whether all slaves shall be reset in case of an error event on a mandatory slave (default: false, see bit 4 in object 1F80).
   stop_all_nodes;	Specifies whether all slaves shall be stopped in case of an error event on a mandatory slave (default: false, see bit 6 in object 1F80).
   boot_time;	The timeout for booting mandatory slaves in ms (default: 0, see object 1F89).
+  boot_timeout;	The timeout for booting all slaves in ms (default: 100ms).
```
Comment (ipa-cmh) [inline canopen/sphinx/user-guide/configuration.rst:114]: 100ms seems very short fot a default. I would say go with something like 2 seconds.

File: canopen_core/include/canopen_core/node_interfaces/node_canopen_master.hpp [hunk 2]
```diff
@@ -166,7 +167,23 @@ class NodeCanopenMaster : public NodeCanopenMasterInterface
...
+    std::optional<int> timeout;
+    try
+    {
+      timeout = this->config_["boot_timeout"].as<int>();
+    }
+    catch (...)
+    {
+      RCLCPP_WARN(
+        this->node_->get_logger(),
+        "No timeout parameter found in config file. Using default value of 100ms.");
```
Comment (ipa-cmh) [inline canopen_core/include/canopen_core/node_interfaces/node_canopen_master.hpp:181]: Change to boot timeout, otherwise it is hard to get.

**Review by ipa-cmh — COMMENTED** [review]
@ipa-vsp can you do these changes?

**Comment by roncapat** [comment]
Any update on this?

File: canopen_core/include/canopen_core/node_interfaces/node_canopen_master.hpp (hunk 2 above)
Comment (Timple) [reply canopen_core/include/canopen_core/node_interfaces/node_canopen_master.hpp:181]: Fixed in: https://github.com/ros-industrial/ros2_canopen/pull/385

**Pull request merged**